addons/mrp_subcontracting/report/mrp_report_bom_structure.py

```diff
@@ -30,6 +30,12 @@ def _get_bom_data(self, bom, warehouse, product=False, line_qty=False, bom_line=
                 res['subcontracting'] = self._get_subcontracting_line(bom, seller, level + 1, res['quantity'])
                 if not self.env.context.get('minimized', False):
                     res['bom_cost'] += res['subcontracting']['bom_cost']
+                    if bom.product_uom_id != product.uom_id:
+                        price = bom.product_uom_id._compute_price(res['subcontracting']['bom_cost'], product.uom_id)
+                        res['bom_unit_cost'] += price
+                    else:
+                        res['bom_unit_cost'] += res['subcontracting']['bom_cost']
```

Comment:
```suggestion
                    res['bom_unit_cost'] += bom.product_uom_id._compute_price(res['subcontracting']['bom_cost'], product.uom_id)
```
`_compute_price()` checks if both uoms are the same anyway, and if they are just returns the price.